### PR TITLE
feat(ffi): add support for FFI interface

### PR DESCRIPTION
**Pull request by Manuthor**

### CODE REVIEW

File: crates/findex/Cargo.toml [hunk 1]
```diff
@@ -72,49 +73,46 @@ backend-wasm = [
...
+redis = { version = "0.23", features = [
+  "aio",
+  "ahash",
+  "script",
+  "connection-manager",
+  "tokio-comp",
+], optional = true }
+reqwest = { version = "0.11.22", features = [
+  "rustls-tls",
+], default-features = false, optional = true }
```
Comment (tbrezot) [inline crates/findex/Cargo.toml:100]: Do we really need TLS here? All Findex communication is encrypted by design.
```suggestion
reqwest = { version = "0.11.22", default-features = false, optional = true }
```

File: crates/findex/src/backends/custom/ffi/stores.rs [hunk 2]
```diff
@@ -121,21 +138,33 @@ macro_rules! impl_ffi_edx_store {
...
+            #[tracing::instrument(level = "trace", fields(map = %map), ret, err, skip(self))]
             async fn insert(
                 &self,
-                map: HashMap<Self::Token, EncryptedValue<$value_length>>,
+                map: cosmian_findex::TokenToEncryptedValueMap<$value_length>,
             ) -> Result<(), Self::Error> {
+                tracing::debug!("insert: entering: map size: {}", map.len());
+                if map.is_empty() {
+                    return Ok(());
+                }
```
Comment (tbrezot) [inline crates/findex/src/backends/custom/ffi/stores.rs:167]: My take on this recently is that empty requests are not the nominal case. They may even be considered as application logic error. Adding tests which sole purpose is to speed up this class of requests is therefore rather hurtful imho: it hurts readability and adds a branching.

I know some of these tests are still around, but I am planning to remove them as I see them.

```suggestion
```

File: crates/findex/src/backends/custom/ffi/stores.rs [hunk 3]
```diff
@@ -144,24 +173,46 @@ macro_rules! impl_ffi_edx_store {
...
-                uids: std::collections::HashSet<Self::Token>,
-            ) -> Result<(), Self::Error> {
-                let serialized_uids = serialize_uid_set(&uids)?;
-                let serialized_uids_len = <u32>::try_from(serialized_uids.len())?;
+            #[tracing::instrument(level = "trace", fields(tokens = %tokens), ret, err, skip(self))]
+            async fn delete(&self, tokens: cosmian_findex::Tokens) -> Result<(), Self::Error> {
+                tracing::debug!("delete: entering: tokens number {}", tokens.len());
+                if tokens.is_empty() {
+                    return Ok(());
+                }
```
Comment (tbrezot) [inline crates/findex/src/backends/custom/ffi/stores.rs:185]:
```suggestion
```

**Review by tbrezot — COMMENTED** [review]
Great job

File: crates/findex/Cargo.toml (hunk 1 above)
Comment (Manuthor) [reply crates/findex/Cargo.toml]: feature removed

**Pull request merged**